packages/@aws-cdk/example-construct-library/lib/example-resource.ts

```diff
@@ -0,0 +1,501 @@
+      detail: {
+        'example-resource-name': [this.exampleResourceName],
+      },
+    });
+    return rule;
+  }
+
+  /** Implement the {@link IExampleResource.metricSth} method. */
+  public metricSth(metricName: string, props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return new cloudwatch.Metric({
+      // of course, you would put your resource-specific values here
+      namespace: 'AWS/ExampleResource',
+      dimensions: { ExampleResource: this.exampleResourceName },
+      metricName,
+      ...props,
+    }).attachTo(this);
+  }
+}
+
+/**
+ * Construction properties for {@link ExampleResource}.
+ * All constructs have the same construction pattern:
+ * you provide a scope of type Construct,
+ * a string identifier, and a third argument,
+ * representing the properties specific to that resource.
+ * That third type is represented in the CDK by an interface
+ * with only readonly simple properties (no methods),
+ * sometimes called, in JSII terminology, a 'struct'.
+ * This is this struct for the {@link ExampleResource} class.
+ *
+ * This interface is always called '<ResourceName>Props'.
+ */
+export interface ExampleResourceProps {
+  /**
+   * The physical name of the resource.
+   * If you don't provide one, CloudFormation will generate one for you.
+   * Almost all resources, with only a few exceptions,
+   * allow setting their physical name.
+   *
+   * @default - CloudFormation-generated name
+   */
+  readonly exampleResourceName?: string;
+
+  /**
+   * Many resources require an IAM Role to function.
+   * While a customer can provide one,
+   * the CDK will always create a new one
+   * (with the correct assumeRole service principal) if it wasn't provided.
+   *
+   * @default - a new Role will be created
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * Many resources allow passing in an optional S3 Bucket.
+   * Buckets can also have KMS Keys associated with them,
+   * so any encryption settings in your resource should check
+   * for the presence of that property on the passed Bucket.
+   *
+   * @default - no Bucket will be used
+   */
+  readonly bucket?: s3.IBucket;
+
+  /**
+   * Many resources can be attached to a VPC.
+   * If your resource cannot function without a VPC,
+   * make this property required -
+   * do NOT make it optional, and then create a VPC implicitly!
+   * This is different than what we do for IAM Roles, for example.
+   *
+   * @default - no VPC will be used
+   */
+  readonly vpc?: ec2.IVpc;
+
+  /**
+   * Whenever you have IVpc as a property,
+   * like we have in {@link vpc},
+   * you need to provide an optional property of type ec2.SubnetSelection,
+   * which can be used to specify which subnets of the VPC should the resource use.
+   * The default is usually all private subnets,
+   * however you can change that default in your resource if it makes sense
+   * (for example, to all public subnets).
+   *
+   * @default - all private subnets
```

Comment:
In fact the default is "default subnet selection strategy, see ec2 module"

Reply:
Done.